src/interpreter/initialization.rs

```diff
@@ -64,9 +74,10 @@ where
     ///
     /// For predicate verification, check [`Self::init`]
     pub fn init_with_storage(&mut self, tx: CheckedTransaction) -> Result<(), InterpreterError> {
-        let predicate = false;
         let block_height = self.storage.block_height().map_err(InterpreterError::from_io)?;
 
-        self.init(predicate, block_height, tx)
+        self.context = Context::Script;
```

Comment:
[nit] Should we also have a `Create` context?

Reply:
Most definitely. Making some tests now to see if its a small impact change